tree_format/_text.py

```diff
@@ -20,66 +20,90 @@
+class Options(object):
 
+    def __init__(self,
+                 FORK=u'\u251c',
+                 LAST=u'\u2514',
+                 VERTICAL=u'\u2502',
+                 HORIZONTAL=u'\u2500',
+                 NEWLINE=u'\u23ce'):
+        self.FORK = FORK
+        self.LAST = LAST
+        self.VERTICAL = VERTICAL
+        self.HORIZONTAL = HORIZONTAL
+        self.NEWLINE = NEWLINE
 
-def _format_newlines(prefix, formatted_node):
+
+def _format_newlines(prefix, formatted_node, options):
     """
     Convert newlines into U+23EC characters, followed by an actual newline and
     then a tree prefix so as to position the remaining text under the previous
     line.
     """
     replacement = u''.join([
-        NEWLINE,
+        options.NEWLINE,
         u'\n',
         prefix])
     return formatted_node.replace(u'\n', replacement)
 
 
-def _format_tree(node, format_node, get_children, prefix=u''):
+def _format_tree(node, format_node, get_children, options, prefix=u''):
     children = list(get_children(node))
-    next_prefix = u''.join([prefix, VERTICAL, u'   '])
+    next_prefix = u''.join([prefix, options.VERTICAL, u'   '])
     for child in children[:-1]:
         yield u''.join([prefix,
-                        FORK,
-                        HORIZONTAL,
-                        HORIZONTAL,
+                        options.FORK,
+                        options.HORIZONTAL,
+                        options.HORIZONTAL,
                         u' ',
                         _format_newlines(next_prefix,
-                                         format_node(child))])
+                                         format_node(child),
+                                         options)])
         for result in _format_tree(child,
                                    format_node,
                                    get_children,
+                                   options,
                                    next_prefix):
             yield result
     if children:
         last_prefix = u''.join([prefix, u'    '])
         yield u''.join([prefix,
-                        LAST,
-                        HORIZONTAL,
-                        HORIZONTAL,
+                        options.LAST,
+                        options.HORIZONTAL,
+                        options.HORIZONTAL,
                         u' ',
                         _format_newlines(last_prefix,
-                                         format_node(children[-1]))])
+                                         format_node(children[-1]),
+                                         options)])
         for result in _format_tree(children[-1],
                                    format_node,
                                    get_children,
+                                   options,
                                    last_prefix):
             yield result
 
 
-def format_tree(node, format_node, get_children):
+def format_tree(node, format_node, get_children, options=Options()):
```

Comment:
Since `Options` instances are technically mutable, this should instead default to `None` and then instantiate `Options()` if `None` was provided.

Alternatively, use attrs and make it an immutable structure.